platforms/core-runtime/files/src/test/groovy/org/gradle/internal/file/FileHierarchySetTest.groovy

```diff
@@ -358,6 +359,19 @@ class FileHierarchySetTest extends Specification {
         rootsOf(from(dir1, commonDir2, commonDir3)) == [dir1.absolutePath, commonDir2.absolutePath, commonDir3.absolutePath]
     }
 
+    def "handles two unrelated paths correctly"() {
+        def set = FileHierarchySet.empty()
+        def absolutePaths
+        if (OperatingSystem.current().isUnix()) {
+            absolutePaths = ['/some/dIR', '/another/dIR']
+        } else {
+            absolutePaths = ['C:\\Some\\DIR', 'D:\\Another\\DIR']
+        }
+
+        expect:
+        rootsOf(fromPaths(absolutePaths)) == absolutePaths
+    }
```

Comment:
:x: This is a unit test, it should test both cases and not rely on `OperatingSystem`.